DataFormats/Provenance/src/BranchType.cc

```diff
@@ -16,52 +20,37 @@ namespace edm {
     std::string const branchEntryInfo = "BranchEntryInfo";  // backward compatibility
     std::string const productProvenance = "ProductProvenance";
 
-    // Prefixes
-    std::string const run = "Run";
-    std::string const lumi = "LuminosityBlock";
-    std::string const event = "Event";
-
-    // Trees, branches, indices
-    std::string const runs = run + 's';
-    std::string const lumis = lumi + 's';
-    std::string const events = event + 's';
+    std::array<std::string, NumBranchTypes> const branchTypeNames{{"Event", "LuminosityBlock", "Run", "ProcessBlock"}};
+    std::size_t constexpr eventLumiRunSize = 3;
+    using NameArray = const std::array<std::string, eventLumiRunSize>;
 
-    std::string const runMeta = run + metaData;
-    std::string const lumiMeta = lumi + metaData;
-    std::string const eventMeta = event + metaData;
+    NameArray makeNameArray(std::string const& postfix) {
+      static_assert(InEvent == 0);
+      static_assert(InLumi == 1);
+      static_assert(InRun == 2);
+      return NameArray{
+          {branchTypeNames[InEvent] + postfix, branchTypeNames[InLumi] + postfix, branchTypeNames[InRun] + postfix}};
+    }
 
-    std::string const runInfo = run + "StatusInformation";      // backward compatibility
-    std::string const lumiInfo = lumi + "StatusInformation";    // backward compatibility
-    std::string const eventInfo = event + "StatusInformation";  // backward compatibility
+    NameArray treeNames{makeNameArray(std::string("s"))};
```

Comment:
A rather minor point, the construction of `std::string` could be avoided if `makeNameArray` would take `std::string_view`.

Reply:
I tried and had difficulties because operator+ does not support adding a string and string_view. This seemed the simplest way around that restriction.